framework/validators/AtLeastOneValidator.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\validators;
+
+use Yii;
+use yii\base\InvalidConfigException;
+
+/**
+ * AtLeastValidator checks if at least one of many attributes are filled.
+ *
+ * In the following example, the `attr1`, `attr2` and `attr3` attributes will
+ * be verified. If at least one of them are not filled, all will receive an error:
+ *
+ * ```php
+ *  ['attr1', 'atLeastOne', 'alternativeAttributes' => ['attr2', 'attr3']],
+ * ```
+ *
+ * In the following example, the `attr1`, `attr2` and `attr3` attributes will
+ * be verified. If at least two of them are not filled, `attr1` and `attr3` will
+ * receive an error:
+ *
+ * ```php
+ *  ['attr1', 'atLeastOne', 'alternativeAttributes' => ['attr2', 'attr3'], 'min' => 2, 'errorAttributes' => ['attr1', 'attr3']],
+ * ```
+ *
+ * Important: Only one attribute should be specified in the attributes.
+ * The additional ones must go into [[alternativeAttributes]] param.
+ *
+ * The attributes that will receive the error message can be specified into [[errorAttributes]]
+ * param. If none is specified, all envolved attributes will receive the error.
+ *
+ * @author Sidney Lins <slinstj@gmail.com>
+ * @since 2.0.14
+ */
+class AtLeastOneValidator extends Validator
+{
+    /**
+     * @var array|string the required list of aditional attributes in which to check
+     * at least one filled.
+     */
+    public $alternativeAttributes;
+    /**
+     * @var integer the minimun required quantity of attributes that must to be filled.
+     * Defaults to 1.
+     */
+    public $min = 1;
+    /**
+     * @var array the list of attributes that should receive the error message.
+     * Defaults to all attributes being validated (i.e, attribute + [[alternativeAttributes]] attributes).
+     */
+    public $errorAttributes;
+    /**
+     * @var boolean whether this validation rule should be skipped if the attribute value
+     * is null or an empty string.
+     */
+    public $skipOnEmpty = false;
+
+    /**
+     * @inheritdoc
+     */
+    public function init()
+    {
+        parent::init();
+        if ($this->isEmpty($this->alternativeAttributes)) {
+            throw new InvalidConfigException('Param "alternativeAttributes" can not be null');
+        }
+        if (count($this->attributes) > 1) {
```

Comment:
\yii\validators\Validator::$attributes can be a string according to its phpdoc.
When using `count`, this will fail on PHP 7.2